src/main/java/com/google/devtools/build/lib/remote/RemoteActionFileSystem.java

```diff
@@ -101,7 +111,9 @@ void injectTree(SpecialArtifact tree, TreeArtifactValue metadata) {
   void injectFile(ActionInput file, RemoteFileArtifactValue metadata) {
     checkNotNull(metadataInjector, "metadataInject is null");
 
-    injectedMetadata.put(file.getExecPath(), metadata);
+    if (metadata instanceof RemoteFileArtifactValue) {
```

Comment:
always true?

Reply:
Oops, it was a leftover from a previous refactor.